### ISSUE TITLE
Spleen_segmentation_3d.ipynb was executed and the predicted label and background pixel values were reversed

### ISSUE THREAD

**Issue by ljljlj02** [Project-MONAI/tutorials#433]
Hello, Monai is a very good job, but I encountered a problem when learning the tutorial. I used Spleen data, ran spleen_segmentation_3D.ipynb, the code did not change, and the Dice of the training set was the same as the tutorial. Dice can achieve 0.94. But there is a problem in the validation set:

In the 'Evaluation on original Image Spacings' module, the tutorial results in:  Metric on original image Spacings: 0.9637420177459717 and MY result is "Metric on original Image spacing: 0.05605338141322136"

In addition, I print out the predicted results, Using `SaveImaged(keys="pred",`` meta_keys="pred_meta_dict", output_dir=out_root_dir,resample=False,Output_postfix `="seg")` ,the result is as follows
![1](https://user-images.githubusercontent.com/50227846/140000677-1b19ead7-908b-4dc2-8a36-82548be90379.png)
![2](https://user-images.githubusercontent.com/50227846/140000681-c1087d3f-2387-48f3-ad0f-6269e2b2a0fb.png)

The result shows that the pixel values of the label and background are opposite, I wonder how to solve this problem.
Any help would be greatly appreciated.

**Comment by Nic-Ma** [comment]
Hi @yiheng-wang-nv ,

Could you please help double confirm the issue as we added this evaluation part several months ago?

Thanks in advance.

**Comment by yiheng-wang-nv** [comment]
hi @ljljlj02 , I double checked the code and rerun the script, but I did not meet the "reversed" issue. You said you did not change any code, that's weird and can you ensure that the weights loaded are correct? By the way, what is the output of `print_config()` at the beginning of the notebook?

**Comment by ljljlj02** [comment]
> hi @ljljlj02 , I double checked the code and rerun the script, but I did not meet the "reversed" issue. You said you did not change any code, that's weird and can you ensure that the weights loaded are correct? By the way, what is the output of `print_config()` at the beginning of the notebook?

Thank you, I don't know how to check my weight, so I have uploaded my execution code and weight in my repository, could you please help me to check where the problem is? (Just execute the parts that I didn't comment out.), my repository [https://github.com/ljljlj02/A-problem-about-Spleen_segmentation_3d]. 

In addition, my `print_config()` are as follows:
`MONAI version: 0.6.0
Numpy version: 1.19.2
Pytorch version: 1.5.0
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: 0ad9e73639e30f4f1af5a1f4a45da9cb09930179

Optional dependencies:
Pytorch Ignite version: 0.4.5
Nibabel version: 3.2.1
scikit-image version: 0.15.0
Pillow version: 8.3.1
Tensorboard version: 2.8.0a20211101
gdown version: NOT INSTALLED or UNKNOWN VERSION.
TorchVision version: 0.6.0
ITK version: NOT INSTALLED or UNKNOWN VERSION.
tqdm version: 4.62.3
lmdb version: NOT INSTALLED or UNKNOWN VERSION.
psutil version: NOT INSTALLED or UNKNOWN VERSION.
pandas version: 1.1.5
einops version: NOT INSTALLED or UNKNOWN VERSION.`

**Comment by rijobro** [comment]
It would be useful to `imshow` your results in the notebook. If they appear normal, that would imply that the problem occurs when saving/loading the image in ITKSnap. If the images appear inverted in the `imshow`, then that implies that something is going wrong elsewhere.

**Comment by ljljlj02** [comment]
I would understand if it was the ITK-Snap display error, but I did not get the same result as your tutorial when EXECUTING the code below:`print("Metric on original image spacing: ", metric_org)`
The tutorial result is "Metric on original image Spacings: 0.9637420177459717 "
My result is "Metric on original Image spacing: 0.05605338141322136"
There is a big difference between 0.9637420177459717 and 0.05605338141322136, which makes me feel that it is not just the problem shown by ITK-SNAP. What is the output result of executing the code in my repository?
Any answers would be greatly appreciated.

**Comment by yiheng-wang-nv** [comment]
Hi @ljljlj02 , I ran your uploaded py file with your trained weights (what I changed are just modify the path variables and convert the file into .ipynb format ), and the result is reasonable: `0.9568932056427002 `

I'm not sure but I guess the weird 0.05+ result may due to the out-dated MONAI or other libraries' versions. The environment I used is MONAI docker:
![Screen Shot 2021-11-04 at 2 10 10 PM](https://user-images.githubusercontent.com/68361391/140266500-24137ccb-31f6-448a-b1a3-eb7ec1233cbf.png)
If possible, can you try with the latest MONAI docker? Or updated the MONAI version?

I've attached the notebook I run, and you can check the printed info there.

[Spleen_segmentation_3d_repository_issue (1).ipynb.zip](https://github.com/Project-MONAI/tutorials/files/7472701/Spleen_segmentation_3d_repository_issue.1.ipynb.zip)
)

**Comment by ljljlj02** [comment]
Thank you very much @yiheng-wang-nv , I updated the versions of PyTorch and Monai with the following results:`Metric on original image spacing:  0.9568932056427002`

My confiure:
`MONAI version: 0.7.0
Numpy version: 1.19.2
Pytorch version: 1.10.0+cu113
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: bfa054b9c3064628a21f4c35bbe3132964e91f43

Optional dependencies:
Pytorch Ignite version: 0.4.5
Nibabel version: 3.2.1
scikit-image version: 0.15.0
Pillow version: 8.3.1
Tensorboard version: 2.8.0a20211101
gdown version: NOT INSTALLED or UNKNOWN VERSION.
TorchVision version: 0.11.1
tqdm version: 4.62.3
lmdb version: NOT INSTALLED or UNKNOWN VERSION.
psutil version: NOT INSTALLED or UNKNOWN VERSION.
pandas version: 1.1.5
einops version: NOT INSTALLED or UNKNOWN VERSION.
transformers version: NOT INSTALLED or UNKNOWN VERSION.`

But the display problem in ITK-Snap was not solved, and I found that the shape of the predicted label was like this:`(2, 135, 512, 512)`, the label has two channels.

Then I saved and output the images of the two channels in as.niI.gz format respectively and checked them in ITK-SNAP. I found that Channel0 was abnormal and Channel 1 was normal, as shown in the screenshot below
![1](https://user-images.githubusercontent.com/50227846/140447150-f84a513b-65ff-41fe-9740-3f727be53f12.png)

I'm not sure if this is a problem with Monai using 'SaveImaged' to save the image. I look forward to your further work.

Best wishes,
ljljlj02

**Comment by ljljlj02** [comment]
> Hello, has this issue been resolved? I'm getting the same problem now. The foreground and background of the results predicted by monai are reversed, and the positions between the predicted results and the original data are not registered.
> 
> The MONAI version information is as follows:
> 
> MONAI version: 0.8.1+135.gd6327a7.dirty Numpy version: 1.19.2 Pytorch version: 1.7.1 MONAI flags: HAS_EXT = False, USE_COMPILED = False MONAI rev id: d6327a765cf23dafd680b2a4b472edf9639e1087
> 
> Optional dependencies: Pytorch Ignite version: 0.4.8 Nibabel version: 3.2.1 scikit-image version: 0.17.2 Pillow version: 8.1.0 Tensorboard version: 2.5.0 gdown version: 4.3.1 TorchVision version: 0.8.2 tqdm version: 4.56.0 lmdb version: 1.3.0 psutil version: 5.9.0 pandas version: 1.1.5 einops version: 0.4.0 transformers version: 4.16.2 mlflow version: 1.23.1
> 
> ![捕获](https://user-images.githubusercontent.com/25079720/157883160-bf09b8ea-042e-4a2c-acfc-6489ec64f737.PNG)
![image](https://user-images.githubusercontent.com/50227846/157998950-ba4cfd47-7523-47f3-8c38-b76a19fcaa0a.png)
I solved this problem, but needed to change the channel each time the result of the forecast output showed the first channel, the correct result was on the second channel, the second channel can be seen in ITK-SNAP, as shown by the drop-down arrow in the image. This problem could also be caused by my "Compose" setup, but there is no solution to this problem except to transform the result each time.

**Comment by sunshinehui** [comment]
> > Hello, has this issue been resolved? I'm getting the same problem now. The foreground and background of the results predicted by monai are reversed, and the positions between the predicted results and the original data are not registered.
> > The MONAI version information is as follows:
> > MONAI version: 0.8.1+135.gd6327a7.dirty Numpy version: 1.19.2 Pytorch version: 1.7.1 MONAI flags: HAS_EXT = False, USE_COMPILED = False MONAI rev id: d6327a765cf23dafd680b2a4b472edf9639e1087
> > Optional dependencies: Pytorch Ignite version: 0.4.8 Nibabel version: 3.2.1 scikit-image version: 0.17.2 Pillow version: 8.1.0 Tensorboard version: 2.5.0 gdown version: 4.3.1 TorchVision version: 0.8.2 tqdm version: 4.56.0 lmdb version: 1.3.0 psutil version: 5.9.0 pandas version: 1.1.5 einops version: 0.4.0 transformers version: 4.16.2 mlflow version: 1.23.1
> > ![捕获](https://user-images.githubusercontent.com/25079720/157883160-bf09b8ea-042e-4a2c-acfc-6489ec64f737.PNG)
> > ![image](https://user-images.githubusercontent.com/50227846/157998950-ba4cfd47-7523-47f3-8c38-b76a19fcaa0a.png)
> > I solved this problem, but needed to change the channel each time the result of the forecast output showed the first channel, the correct result was on the second channel, the second channel can be seen in ITK-SNAP, as shown by the drop-down arrow in the image. This problem could also be caused by my "Compose" setup, but there is no solution to this problem except to transform the result each time.

Yeah, very useful for me, thank you very much. It's just strange why the predicted results has two channels, which adds tediousness to the subsequent operations
